appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/rest/MetricsResource.java

```diff
@@ -142,13 +143,38 @@ private MetricsWriter getOutputWriter(HttpServletRequest request, HttpServletRes
 
         switch (method) {
             case GET:
-                if (accept.contains(APPLICATION_JSON)) {
+                //application/json;q=0.1,text/plain;q=0.9
+
+                String[] acceptFormats = accept.split(",");
+                float qJsonValue = 0;
+                float qTextFormat = 0;
+                for (String format : acceptFormats) {
+                    if (format.contains(TEXT_PLAIN) || format.contains(MediaType.WILDCARD) || format.contains("text/*")) {
+                        String[] splitTextFormat = format.split(";");
+                        if (splitTextFormat.length == 2) {
+                            qTextFormat = Float.parseFloat(splitTextFormat[1].substring(2));
+                        } else {
+                            qTextFormat = 1;
+                        }
+                    } else if (format.contains(APPLICATION_JSON) || format.contains("application/*")) {
```

Comment:
`application/*` should not be considered for JSON format otherwise OPTIONS request case should also add this filter.